packages/frontend/src/pages/api/history.ts

```diff
@@ -0,0 +1,58 @@
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+import { db } from "@/server/db";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse,
+) {
+  if (req.method === "GET") {
+    try {
+      const { user_id } = req.query;
+
+      const searchHistory = await db.searchHistory.findMany({
+        where: {
+          user_id: user_id as string,
+        },
+      });
+      res.status(200).json(searchHistory);
+    } catch (err) {
+      res.status(500).json({ error: err });
+    }
+  }
+
+  if (req.method === "POST") {
+    try {
+      const { user_id, search_content, search_category } = JSON.parse(
+        req.body as string,
+      );
+      const today = new Date();
+      today.setHours(0, 0, 0, 0);
+      const searchHistory = await db.searchHistory.findMany({
+        where: {
+          user_id: user_id as string,
+          search_content: search_content as string,
+          search_category: search_category as string,
+          created_at: { gte: today },
```

Comment:
why pass in date when the date has a default db generated `now()`?

Reply:
bc i want to setHours(0, 0, 0, 0), and Date.now() can't do that